pkg/controller/hosted/controller.go

```diff
@@ -174,7 +171,25 @@ func (r *ReconcileHosted) importCluster(ctx context.Context,
 			nil
 	}
 
-	hostedWorks, err := r.informerHolder.HostedWorkLister.ManifestWorks(hostingCluster).List(hostedWorksSelector)
+	hostingCluster := &clusterv1.ManagedCluster{}
+	err = r.clientHolder.RuntimeClient.Get(ctx, types.NamespacedName{Name: hostingClusterName}, hostingCluster)
+	if err != nil {
+		return reconcile.Result{},
+			helpers.NewManagedClusterImportSucceededCondition(metav1.ConditionFalse,
+				constants.ConditionReasonManagedClusterImporting,
+				fmt.Sprintf("Validate if the hosting cluster is a managed cluster of the hub, error: %v", err)),
```

Comment:
nit: this message may be confused, how about use failed to get the managed cluster %s

Reply:
And we may only need to focus the notfound error,  for other errors, when the errors happed, we will update condition, so the same error may be also happed for update action